src/signing.rs

```diff
@@ -688,13 +688,18 @@ impl<'d> Deserialize<'d> for SigningKey {
 // better-designed, Schnorr-based signature scheme, see Trevor Perrin's work on
 // "generalised EdDSA" and "VXEdDSA".
 pub(crate) struct ExpandedSecretKey {
+    // `key_bytes` and `key` are separate, because the public key is computed as an unreduced
+    // scalar multiplication (ie `mul_base_clamped`), whereas the signing operations are done
+    // modulo l.
+    pub(crate) key_bytes: [u8; 32],
```

Comment:
I had to add a new field to the expanded secret keys. Ideally, we'd just clamp an integer like the spec demands, and reduce modulo `l`, and let that be the canonical key. In fact, that's what [crrl](https://github.com/pornin/crrl/blob/0dd2fa2369cc8746135807ee82592f9fe07229cc/src/ed25519.rs#L1680) and [Go's ed25519](https://cs.opensource.google/go/go/+/master:src/crypto/internal/edwards25519/scalar.go;drc=da4687923b8c2d42c23f61fa3db9f4d3ce0c5f54;l=204) do.

The reason we don't do that is mainly because it is a breaking change: when `s = clamp(H(seed)[:32])` is an unreduced clamped scalar and `B` is the basepoint, then `s.reduce() * B != s * B`. We don't want people's pubkeys to change overnight, so we keep the old `s * B` computation.